components/engine_rocks/src/ttl_properties.rs

```diff
@@ -67,31 +69,45 @@ impl TablePropertiesCollector for TtlPropertiesCollector {
         if entry_type != DBEntryType::Put {
             return;
         }
-        // only consider data keys
+        // Only consider data keys.
         if !key.starts_with(keys::DATA_PREFIX_KEY) {
             return;
         }
+        // Only consider raw keys.
+        match self.api_version {
```

Comment:
as api_version is fixed during running, can we just use different collectors according to api_version when registering the collector

Reply:
or make collector generic and extract a trait that different ApiVersion implements on its own.

Reply:
with a second thought, maybe it doesn't make much difference considering the current CPU branch prediction.

Reply:
Well, I've been working on it. So far I can say the generic is flooding into everywhere... I'm trying to narrow it into a smaller scope. If I can't make it, I'll take the current version.

Reply:
consider running a micro bench to compare how much it affects the performance

Reply:
I'm going to merge this first and open a new PR for this refactor.